python/ml4ir/base/model/scoring/scorer_factory.py

```diff
@@ -67,8 +67,9 @@ def get_scorer(model_config: dict,
     """
 
     if (MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS in model_config and
-            model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TRIALS,
-                                                                                  0)):
+            (model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TEST_TRIALS, 0) or
+            model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS].get(MonteCarloInferenceKey.NUM_TRAINING_TRIALS, 0))):
```

Comment:
@mohazahran if you use just `num_trials`, it'll simplify all this.

Reply:
same as my previous comment. I prefer being explicit.